src/cards/bloat.ts

```diff
@@ -92,6 +92,7 @@ const spell: Spell = {
       const radiusBoost = unit.modifiers[id]?.radiusBoost;
       const adjustedRadius = getAdjustedRadius(radiusBoost);
       explode(unit, adjustedRadius, damage * quantity, getAdjustedPushDist(radiusBoost),
+        unit.modifiers[id]?.sourceUnit, // TODO - Implement source unit for modifiers
```

Comment:
I would say no. The player unit (or bloat caster) is the one that inflicted the damage here, so should be the one whose onDealDamage events should trigger. Practical examples: The player unit has 50% critical strike chance, or a blessing that makes all of their outgoing damage inflict poison. The player should apply that crit/poison through the bloat, because it is _their_ spell/outgoing damage. The bloated unit is not the one dealing damage, and should not have their own effects trigger.

I found that storing a unit in a modifier leads to all kinds of problems, and doing the id workaround like I did with soul shard is somewhat complicated, so I left it like this for now. The Poison and Bloat modifiers do not actually reference their source unit currently.

Reply:
What's complicated about storing the unit id and fetching the unit object later when it's needed?

Reply:
"Messy" is probably a better word. I ran into some problems with prediction/realtime id's being different than one another, so if you're not careful you can run into some really weird behavior. You don't always have access to the prediction variable either so in some cases it can be tricky to access it (which is why you saw me accessing the unit.isPrediction variable in the Soul Shard PR).

I can definitely do it, it's just a little bit of a workaround and requires a few extra helper functions. I just need to move some of these helper functions to Underworld to make sure it looks clean 👍

Reply:
You can use `predictionUnit.real` to backreference the real unit.
So to find a prediction unit that matches a real unit you can do `predictionUnits.find(pu => pu.real.id == searchID)`.
It's a little expensive of a computation cause it potentially has to loop though all units but that's how it can be done.

Alternatively we could add a reference  to the prediction unit from the real unit.  Like, when copyForPredictionUnit is called, we assign the property `u.predictionUnit = newlyMadePredictionUnit` so the real unit can keep a reference to the current prediction unit.
This may actually be the best solution since I'm going to be adding a sync change that will probably make use of this.
(Note: if you do this you'll have to update the unit serialization function to omit the `predictionUnit` property

Reply:
I'll probably finish the sync update today.  So this new `predictionUnit` property would be available to you tomorrow.  I'd say wait for my change and then this should be an easy update for you

Reply:
Sounds good!

Reply:
Here it is: https://github.com/jdoleary/Spellmasons/pull/577